src/resolve/resolved-dns-dnssec.c

```diff
@@ -445,9 +530,18 @@ static void md_add_uint16(gcry_md_hd_t md, uint16_t v) {
         gcry_md_write(md, &v, sizeof(v));
 }
 
-static void md_add_uint32(gcry_md_hd_t md, uint32_t v) {
+static size_t fwrite_uint8(uint8_t v, FILE *fp) {
```

Comment:
could you swap the two args please? libc is weird that fputc() take the argument first, and the object to operate on second, but we really shouldn't replicate that here...